src/gg.regression.unity.bots/Runtime/Scripts/GenericBots/Experimental/QLearningBot.cs

```diff
@@ -0,0 +1,623 @@
+                {
+                    actionVals[actionEntry.Key] = actionEntry.Value.ToObject<float>();
+                }
+                _qTable.Add(entry.Key, actionVals);
+            }
+            return true;
+        }
+
+        public void Start()
+        {
+            if (!RGActionManager.IsAvailable)
+            {
+                RGDebug.LogError("Action manager is currently unavailable");
+                Destroy(this);
+                return;
+            }
+            DontDestroyOnLoad(this);
+    
+            _validActions = new List<QAction>();
+            _lastInputs = new List<RGActionInput>();
+            _mouseBtnsBuf = new HashSet<int>();
+            _initialSceneName = SceneManager.GetActiveScene().name;
+            _rewardModule = CreateRewardModule();
+            _experienceBuf = new Queue<QExperience>();
+            
+            if (LoadModel())
+            {
+                RGDebug.Log("Loaded existing model with epsilon " + _epsilon);
+            }
+            else
+            {
+                if (!Training)
+                {
+                    Destroy(this);
+                    throw new Exception("No existing model is available");
+                }
+                _qTable = new Dictionary<string, Dictionary<string, float>>();
+                _epsilon = 1.0f;
+            }
+            
+            if (Training)
+            {
+                _speedup = new GameSpeedup(TrainingTimeScale);
+            }
+            
+            OnEpisodeStart();
+        }
+    
+        private void ComputeValidActions()
+        {
+            var validActions = new Dictionary<RGGameAction, List<IRGGameActionInstance>>();
+            foreach (var actionInst in RGActionManager.GetValidActions())
+            {
+                var act = actionInst.BaseAction;
+                List<IRGGameActionInstance> instances;
+                if (!validActions.TryGetValue(act, out instances))
+                {
+                    instances = new List<IRGGameActionInstance>();
+                    validActions.Add(act, instances);
+                }
+                instances.Add(actionInst);
+            }
+    
+            _validActions.Clear();
+            foreach (var qact in _actionSpace)
+            {
+                var act = qact.Action;
+                if (validActions.TryGetValue(act, out var instances))
+                {
+                    qact.CurrentInstances.Clear();
+                    foreach (var inst in instances)
+                    {
+                        if (inst.IsValidParameter(qact.ParamValue))
+                        {
+                            qact.CurrentInstances.Add(inst);
+                        }
+                    }
+                    if (qact.CurrentInstances.Count > 0)
+                    {
+                        _validActions.Add(qact);
+                    }
+                }
+            }
```

Comment:
What exactly is happening here? Is there a reason why ComputeValidActions does not just equal RGActionManager.GetValidActions()?